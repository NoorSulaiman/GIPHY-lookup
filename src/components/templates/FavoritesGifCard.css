#favgifcard{
    transition: transform .1s
}

#favgifcard:hover{
    transform: scale(1.2);
    z-index: 100;
}

#favgifcard:hover #deleteIcon{
    display: block;
}

#deleteIcon:hover{
    background-color:rgb(46, 45, 45); 
}

#deleteIcon:hover #tooltipdelete{
    display: block;
}

#deleteIcon:active #tooltipdelete{
    display: none;
}

#deleteIcon{
    padding: 3%;
    margin: 2%;
    position:absolute;
    right: 0;
    top: 0;
    display: none;
    color: white;
    background-color: rgba(107, 102, 102, 0.52);
    border-radius: 50%;
}

#delIcon{
    margin:0 auto;
}

#tooltipdelete{
    display: none;
    width: 80px;
    background-color: black;
    color: #fff;
    text-align: center;
    border-radius: 6px;
    position: absolute;
    z-index: 110;
}

@media only screen and (max-width:786px){
    img{
        width:100%;
        max-width: 100%;
        overflow: hidden;
        position: relative;
        margin: 0 auto;
    }
    #favgifcard{
        transition: none
    }
    #favgifcard:hover{
        transform: none;
        z-index: unset;
    }
    #favgifcard:active {
        transform: scale(0.9,0.9) translateY(2px)
    }
    #deleteIcon{
    margin: 1% auto;
    width:11.5%;
    padding: 3%;
    position: unset;;
    display: block;
    color: white;
    background-color:rgb(46, 45, 45); 
    border-radius: 50%;
    }
    #deleteIcon:hover{
        
    }
}