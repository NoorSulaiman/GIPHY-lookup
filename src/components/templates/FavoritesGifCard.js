import React from 'react';
import PropTypes from 'prop-types';
import { Card, Icon } from 'semantic-ui-react';
import './FavoritesGifCard.css';

const FavoritesGifCard = ({ title, imgUrl, id, removeFunc }) => (
    <Card id='favgifcard'>
        <img alt="gifImage" src={imgUrl} />
        {/* todo: switch to button */}
        <Icon as='a' id='deleteIcon' onClick={() => removeFunc(id)}>
            <Icon size='large' name='close' id='delIcon' />
            <span id="tooltipdelete">Delete!</span>
        </Icon>

        <Card.Content>
            <Card.Header>{title}</Card.Header>
        </Card.Content>
    </Card>
);


FavoritesGifCard.propTypes = {
    title: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    removeFunc: PropTypes.func.isRequired
}

export default FavoritesGifCard;