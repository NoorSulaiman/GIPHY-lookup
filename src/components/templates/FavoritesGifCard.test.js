import React from 'react';
import { shallow } from 'enzyme';
import FavoritesGifCard from './FavoritesGifCard';

const clickFn = jest.fn();
const title = 'title';
const imgUrl = 'url';
const id = 'id';

it('should render FavoritesGifCard correctly with given props', () => {
    const component = shallow(<FavoritesGifCard title={title} imgUrl={imgUrl} id={id} removeFunc={clickFn} />);
    expect(component).toMatchSnapshot();
    component
        .find('#deleteIcon')
        .simulate('click');
    expect(clickFn).toHaveBeenCalled();
});
