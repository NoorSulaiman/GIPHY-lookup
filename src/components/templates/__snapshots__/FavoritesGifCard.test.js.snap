// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FavoritesGifCard correctly with given props 1`] = `
<Card
  id="favgifcard"
>
  <img
    alt="title"
    src="url"
  />
  <Icon
    as="a"
    id="deleteIcon"
    onClick={[Function]}
  >
    <Icon
      as="i"
      id="delIcon"
      name="close"
      size="large"
    />
    <span
      id="tooltipdelete"
    >
      Delete!
    </span>
  </Icon>
  <CardContent>
    <CardHeader>
      title
    </CardHeader>
  </CardContent>
</Card>
`;
